packages/eslint-rules/src/rules/find_nested_nodes.ts

```diff
@@ -0,0 +1,50 @@
+import { TSESTree } from '@typescript-eslint/utils';
+
+/**
+ * Generator function that runs BFS on nodes in catch block AST
+ */
+//eslint-disable-next-line no-restricted-syntax
+export function* findNestedNodes(
+  root: TSESTree.CatchClause,
+  addToCauseNames: (name: string) => void
+) {
+  const queue = [];
+  let body = root.body.body[0];
+
+  if (body.type === 'VariableDeclaration') {
+    for (const newBody of root.body.body) {
+      if (newBody.type === 'VariableDeclaration') {
+        addToCauseNames((body.declarations[0].id as TSESTree.Identifier).name);
```

Comment:
This is a very use case specific logic.

I'd bubble this up to caller and drop `addToCauseNames` parameter.

Btw. I think we can handle this with single for loop at the caller. Because presumably variable declarations are always before their usage, so populating `causeVarNames` and processing other conditions at the same time should work fine.